### ISSUE TITLE
`smartled` feature shouldn't be activated by default

### ISSUE THREAD

**Issue by bjoernQ** [esp-rs/esp-hal#105]
Currently the feature is always activated as soon as a target is activated (which is always the case).

This feature must be controllable by a feature of the chip-specific HALs and not activated by default.

The example should not be build when the feature is not activated. That should be doable by something like this

```toml
[[example]]
name = "hello_rgb"
required-features = ["smartled"]
```

**Comment by jessebraham** [comment]
I agree, however this unfortunately will complicate building the examples in CI.

**Comment by MabezDev** [comment]
Hmm this will also need to be solved before #103 can be merged, as default `vectored` features will need to be added to the chip hals, not the common one.
